clients/redigo.go

```diff
@@ -0,0 +1,389 @@
+package clients
+
+import (
+	"fmt"
+	"github.com/Shivam010/go-rejson/rjs"
+	redigo "github.com/gomodule/redigo/redis"
+	"strings"
+)
+
+// Redigo implements ReJSON interface for GoModule/Redigo Redis client
+// Link: https://github.com/gomodule/redigo
+type Redigo struct {
+	Conn redigo.Conn // import redigo "github.com/gomodule/redigo"
+}
+
+// JSONSet used to set a json object
+//
+// ReJSON syntax:
+// 	JSON.SET <key> <path> <json>
+// 			 [NX | XX]
+//
+func (r *Redigo) JSONSet(key string, path string, obj interface{}, opts ...rjs.SetOption) (
+	res interface{}, err error) {
+
+	if len(opts) > 1 {
+		return nil, rjs.TooManyOptionals
+	}
+	args := make([]interface{}, 0, 5)
+	args = append(args, key, path, obj)
+
+	if len(opts) == 1 {
+		args = append(args, opts[0].Value()...)
+	}
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_SET, args...)
```

Comment:
> rjs.ReJSONCommand_SET

Not sure how I feel about mixing underscores with Camel-case, but I suppose its ok for now.

Reply:
It is not, general that the under_score gives more clarity over camelCase, it's just a perspective of thinking. 
We can remove it too, I have no problem.